### PR TITLE
DD-919: add citation year to dataset metadata

### PR DESCRIPTION
**Pull request by jo-pol**
Fixes DD-919: citation year

# Description of changes

* add citation year to dataset metadata
* use last FileAccessRequest for both versions of a dataset
* added indexes to tabels

# How to test

* build, deploy and run the three types of commands
* for the indexes: in psql examine the result of `\d expected_*` and \d actual_*

# Related PRs
(Add links)
*

# Notify
@DANS-KNAW/dataversedans

### CODE REVIEW

File: src/main/java/nl/knaw/dans/migration/core/DataverseLoader.java [hunk 1]
```diff
@@ -106,6 +107,11 @@ public void loadFromDataset(String doi) {
             actualDataset.setDoi(shortDoi);
             actualDataset.setDepositor(depositor);
             actualDataset.setFileAccessRequest(v.isFileAccessRequest());
+            v.getMetadataBlocks()
+                .get("citation").getFields().stream()
+                .filter(field -> "dateOfDeposit".equals(field.getTypeName()))
```
Comment (lindareijnhoudt) [inline src/main/java/nl/knaw/dans/migration/core/DataverseLoader.java:113]: is this the date you are using as the citation date? The date i want is the publicationdate, as recorded in the dvobject table

Comment (jo-pol) [reply src/main/java/nl/knaw/dans/migration/core/DataverseLoader.java:113]: For now. 
Reading from the dvndb requires a lot of fuzz like an additional bundle. I prefer a separate PR for that purpose.

File: src/main/java/nl/knaw/dans/migration/core/ExpectedLoader.java [hunk 2]
```diff
@@ -54,6 +54,7 @@ public void expectedMigrationFiles(String doi, String[] migrationFiles, FileRigh
       expectedFile.setRemovedDuplicateFileCount(0);
       expectedFile.setTransformedName(false);
       expectedFile.setDefaultRights(datasetRights);
+      expectedFile.setVisibleTo("ANONYMOUS");
```
Comment (lindareijnhoudt) [inline src/main/java/nl/knaw/dans/migration/core/ExpectedLoader.java:56]: i also want accessibleTo
```expectedFile.setAccessibleTo("ANONYMOUS");```

**Review by lindareijnhoudt — CHANGES_REQUESTED** [review]
i also want accessibleTo on easy-migration files set to anonymous
expectedFile.setAccessibleTo("ANONYMOUS");

File: src/main/java/nl/knaw/dans/migration/core/ExpectedLoader.java (hunk 2 above)
Comment (jo-pol) [reply src/main/java/nl/knaw/dans/migration/core/ExpectedLoader.java:56]: I suppose the embargo-date should also not be set so I'll drop the call to `setDefaultRights`.

**Pull request merged**